File: articles/warren-mcculloch-walter-pitts-and-neural-logic/prolog/threshold_unit.pl
weighted_sum([], [], 0).
weighted_sum([X|Xs], [W|Ws], Total) :-
    weighted_sum(Xs, Ws, Rest),
    Total is X * W + Rest.

threshold_unit(Inputs, Weights, Threshold, Output) :-
    weighted_sum(Inputs, Weights, Total),
    (Total >= Threshold -> Output = 1 ; Output = 0).

% Query example:
% ?- threshold_unit([1,1], [1,1], 2, Output).

File: articles/warren-mcculloch-walter-pitts-and-neural-logic/fortran/threshold_unit.f90
program threshold_unit_example
  implicit none
  integer :: inputs(2), weights(2), total, output
  inputs = (/ 1, 1 /)
  weights = (/ 1, 1 /)
  total = sum(inputs * weights)
  if (total >= 2) then
    output = 1
  else
    output = 0
  end if
  print *, output
end program threshold_unit_example
